hack/jenkins/verify-jenkins-jobs.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+# Copyright 2015 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Pass in Jenkins' URL (localhost:8080 if running this on Jenkins) in $JENKINS.
+# Pass in the location of e2e.sh in $E2E.
+# This will tell you which tests are in one but not the other.
+
+# It would be great if each test project went through the same file.
+# As it stands, build* go through build.sh, so won't be in e2e.sh.
+# check-links goes through verify-linkcheck.sh
+# test-go goes through gotest-dockerized.sh
+# pull* run on the pull Jenkins instance
+readonly EXCEPTIONS='
+kubernetes-build
+kubernetes-build-1.0
+kubernetes-build-1.1
+kubernetes-check-links
+kubernetes-test-go
+kubernetes-pull-build-test-e2e-gce
+kubernetes-pull-test-unit-integration
+kubernetes-verify-jenkins-jobs
+'
+
+# For each element of $1, searches for it in $2. If there are missing elements,
+# print them, labelled by $3.
+# Does not consider builds in EXCEPTIONS or non-kubernetes builds.
+# Returns failure through the FAIL variable.
+function search_build {
+  FAIL=0
+  for build in $1; do
+    grep "^kubernetes-" <(echo "${build}") > /dev/null
```

Comment:
though actually you can probably simplify everything with a bit of pipelining (not fully tested, but the basic jist):

first turn `EXCEPTIONS` into an array:
```shell
EXCEPTIONS=(
  "kubernetes-build"
  "kubernetes-build-1.1"
  ... etc ...
  )

function join_regex() {
  local IFS="|"
  echo "$*"
}
```

and then
```shell
builds_to_consider=grep "^kubernetes-" <(echo "${build}") | grep -v "$(join_regex ${EXCEPTIONS[@]:+${EXCEPTIONS[@]}})"
```

You might be able to even fold the last grep in, though I haven't though about that enough...